File: resources/exercises/array/arraycount9/ArrayCount9Entity.java
package array.arraycount9;
import plm.core.model.lesson.BatEntity;
import plm.core.model.lesson.Lesson;
import plm.universe.bat.BatTest;
import plm.universe.bat.BatWorld;

public class ArrayCount9Entity extends BatEntity {
	public void run(BatTest t) {
		t.setResult( arrayCount9((int[])t.getParameter(0)) );
	}

	/* BEGIN TEMPLATE */
	int arrayCount9(int[] nums) {
		/* BEGIN SOLUTION */
		int count = 0;
		for (int i=0; i<nums.length; i++) {
			if (nums[i] == 9) {
				count++;
			}
		}
		return count;
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: src/lessons/welcome/array/arraycount9/ArrayCount9Entity.py
from RemoteBat import *
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(arrayCount9(param[0])))

# BEGIN TEMPLATE
def arrayCount9(nums):
# BEGIN SOLUTION
  res = 0
  for value in nums:
    if value == 9:
      res += 1
  return res
# END SOLUTION
# END TEMPLATE
